app/collectors/launchTemplates.scala

```diff
@@ -0,0 +1,122 @@
+import scala.util.Try
+
+class LaunchTemplateCollectorSet(accounts: Accounts)
+    extends CollectorSet[LaunchTemplateVersion](
+      ResourceType("launch-templates"),
+      accounts,
+      Some(Regional)
+    ) {
+  val lookupCollector
+      : PartialFunction[Origin, Collector[LaunchTemplateVersion]] = {
+    case amazon: AmazonOrigin =>
+      AWSLaunchTemplateCollector(
+        amazon,
+        resource,
+        amazon.crawlRate(resource.name)
+      )
+  }
+}
+
+case class AsgLaunchTemplate(id: String, version: String)
+
+case class AWSLaunchTemplateCollector(
+    origin: AmazonOrigin,
+    resource: ResourceType,
+    crawlRate: CrawlRate
+) extends Collector[LaunchTemplateVersion]
+    with Logging {
+
+  val asgClient: AutoScalingClient = AutoScalingClient.builder
+    .credentialsProvider(origin.credentials.provider)
+    .region(origin.awsRegionV2)
+    .overrideConfiguration(AWS.clientConfig)
+    .build
+
+  val ec2Client: Ec2Client = Ec2Client.builder
+    .credentialsProvider(origin.credentials.provider)
+    .region(origin.awsRegionV2)
+    .overrideConfiguration(AWS.clientConfig)
+    .build
+
+  def crawl: Iterable[LaunchTemplateVersion] = {
+    val asgs =
+      asgClient.describeAutoScalingGroups().autoScalingGroups().asScala.toList
+    val launchTemplates = asgs
+      // not all asgs have launch templates (they might have launch configurations instead
+      .filter(asg => asg.launchTemplate() != null)
+      .map(asg =>
+        AsgLaunchTemplate(
+          asg.launchTemplate().launchTemplateId(),
+          asg.launchTemplate().version()
+        )
+      )
+
+    launchTemplates.map { lt =>
+      // might need to filter out asgs which don't have launch templates?
+      val requestLt = DescribeLaunchTemplateVersionsRequest.builder
+        .launchTemplateId(lt.id)
+        .versions(lt.version)
+        .build
+      val template = ec2Client
+        .describeLaunchTemplateVersions(requestLt)
+        .launchTemplateVersions()
+        .asScala
+        .head
+      LaunchTemplateVersion.fromApiData(template, origin)
+    }
+
+  }
+}
+
+object LaunchTemplateVersion {
+  def fromApiData(
+      config: AwsLaunchTemplateVersion,
+      origin: AmazonOrigin
+  ): LaunchTemplateVersion = {
+    LaunchTemplateVersion(
+      arn = config.launchTemplateId(),
+      name = config.launchTemplateName(),
+      imageId = config.launchTemplateData().imageId(),
+      region = origin.region,
+      instanceProfile =
+        Option(config.launchTemplateData().iamInstanceProfile().arn()),
+      createdTime = Try(config.createTime()).toOption,
+      instanceType = config.launchTemplateData().instanceType().toString,
+      securityGroups = Option(
+        config.launchTemplateData().securityGroups()
+      ).toList.flatMap(_.asScala).map { sg =>
+        s"arn:aws:ec2:${origin.region}:${origin.accountNumber.get}:security-group/$sg"
+      }
+    )
+  }
+}
+
+case class LaunchTemplateVersion(
```

Comment:
Should we be capturing the tags for the launch template as well? It looks like the tags are available on the launchTemplateData via the `tagSpecifications` field?

Reply:
Unfortunately the tags on the launch template aren't available - we would have to make another API call. `tagSpecifications` is the tags that will be applied to instances launched using this launch template. I'm inclined to leave them out of prism for now until we need them...